### PR TITLE
Fix audit of httpd24 and add support for mod_http2 and mod_ldap

### PR DESCRIPTION
**Pull request by kabel**
Fixes makefile that installs a config file in the "sbin" path.
Fixes layout that installs include headers in shadow system path
Fixes mod_ldap won't build without right dependencies.
Adds allow mod_http2 to be build.

### CODE REVIEW

**Comment by KorvinSzanto** [comment]
Is there currently no other way to enable http2 support with homebrew apache?

File: httpd24.rb [hunk 1]
```diff
@@ -18,12 +18,23 @@ class Httpd24 < Formula
...
 
+  if build.with? "ldap"
+    depends_on "apr-util" => "with-openldap"
+  else
+    depends_on "apr-util"
+  end
+
+  if build.with? "http2"
+    depends_on "nghttp2"
+  end
```
Comment (bfontaine) [inline httpd24.rb]: You can inline this: `depends_on "nghttp2" if build.with? "http2"`

**Comment by kabel** [comment]
Everything else look good? It seems like people are a little anxious to get apache updated to the latest.

**Comment by kabel** [comment]
@bfontaine Any way to get this pushed through?

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by kabel** [comment]
Tests seem to be okay. :shipit: ?

**Comment by bfontaine** [comment]
Yeah it’s for re-generating the bottles; the bot only stores them for 7 days or so. Sorry for the delay, and thank you for your contribution to Homebrew! :tada:

**Comment by peschee** [comment]
@kabel I'm somehow unable to install the httpd24 package with HTTP2 support. I've used this to re-install

```
brew reinstall homebrew/apache/httpd24 --with-brewed-openssl --with-mpm-event --with-http2
```

However, the list of modules shows:

```
/usr/local/bin/apachectl -t -D DUMP_MODULES
Loaded Modules:
 core_module (static)
 so_module (static)
 http_module (static)
 mpm_event_module (static)
 authn_file_module (shared)
 authn_core_module (shared)
 authz_host_module (shared)
 authz_groupfile_module (shared)
 authz_user_module (shared)
 authz_core_module (shared)
 access_compat_module (shared)
 auth_basic_module (shared)
 socache_shmcb_module (shared)
 reqtimeout_module (shared)
 filter_module (shared)
 mime_module (shared)
 log_config_module (shared)
 env_module (shared)
 headers_module (shared)
 setenvif_module (shared)
 version_module (shared)
 ssl_module (shared)
 unixd_module (shared)
 status_module (shared)
 autoindex_module (shared)
 vhost_alias_module (shared)
 dir_module (shared)
 actions_module (shared)
 alias_module (shared)
 fastcgi_module (shared)
```

Any ideas why the HTTP2 module is missing?

**Comment by kabel** [comment]
@peschee It's a shared module, so you'll need to edit your `httpd.conf` file to include a line:
```
LoadModule http2_module libexec/mod_http2.so
```

**Comment by peschee** [comment]
Ah, that was it! Thanks for the prompt reply.

**Pull request closed without merging**